### DISCUSSION TITLE
Import chat history from an older version?

### DISCUSSION THREAD

**Discussion by wooseopkim** [open-webui/open-webui#520] [Q&A] [answered]
I've just updated `ollama` and `ollama-webui` after a while. `ollama-webui` said that I needed to download my chat history because it won't support saving the history in browser anymore. I downloaded it and tried to re-import, but it failed silently. After opening the history file in a text editor and comparing it to the freshly generated one with the new version, it seems like `ollama-webui` doesn't automatically perform schema migration or something. The server returns `422 Unprocessable Entity` and the console error reads `chat.body` is missing.

```
{
    "detail": [
        {
            "type": "missing",
            "loc": [
                "body",
                "chat"
            ],
            "msg": "Field required",
            "input": {},
            "url": "https://errors.pydantic.dev/2.5/v/missing"
        }
    ]
}
```

Do I need to handle it myself? I mean I could, but at the same time I'm just too lazy.

**Comment by wooseopkim** [comment, marked as answer]
I've done a semi-manual migration myself by gradually upgrading then importing again and again, but I've lost my `Modelfile` definitions. They're still on the `ollama` DB, but they won't show up in the `webui` and profile photos are gone. I hope I have no more issues like this in future releases. I sincerely love this app though.

**Comment by tjbck** [comment]
#522 should fix it! Thanks for letting us know!
